defaults/parameters.yaml

```diff
@@ -137,18 +137,29 @@ ancestral:
 # Frequencies settings
 frequencies:
 
-  # min_date is set by default to 1 year before present
-  # but can be explicitly set if desired
+  # default settings that can be over-ridden for specific builds
+  default:
+
+    # min_date is set by default to 1 year before present
+    min_date: "1Y"
+
+    # max_date is set by default to present date - recent_days_to_censor
+
+    # KDE bandwidths in proportion of a year to use per strain.
+    # using 1M bandwidth by default
+    narrow_bandwidth: 0.0833
+
+
+  # settings that can be over-ridden across all builds, but not for specific builds
+  recent_days_to_censor: 0
```

Comment:
In general, I liked having workflow default in addition to the parameters.yaml default. As you point out this redundancy also exists for `frequencies.narrow_bandwidth`. 

But perhaps I understand your concern, where by having multiple layers of "defaults" it might add confusion. It would be nice if people could look just at `parameters.yaml` to understand defaults rather than digging into the workflow.

I think the original push for adding workflow redundancy here was that `parameters.yaml` (still) doesn't include `recent_days_to_censor` and if we start looking for it in the workflow and people are stilling using old `parameters.yaml` files then things will break.

I think we need a broader pattern to adhere to here (kind of like the conversion in issue https://github.com/nextstrain/ncov/issues/1131). I'm not immediately sure what's best for this specific PR.

Reply:
> if we start looking for it in the workflow and people are stilling using old `parameters.yaml` files then things will break.

I'm not familiar with external usage of ncov so I appreciate the thoughts here.

Since `parameters.yaml` is under version control, shouldn't it be sufficient to have workflow changes + `parameters.yaml` changes (adding `recent_days_to_censor`) in the same PR? If someone pulls the workflow changes, they'll also pull the default `recent_days_to_censor` in `parameters.yaml`. I can think of a few possibilities where breakage might happen:

1. User has made custom changes to `parameters.yaml`. `git pull` will try to auto-merge and and flag any merge conflicts.
2. User has explicitly removed this line from Snakefile: https://github.com/nextstrain/ncov/blob/82ca8d48f671349c5688a055bcc67e108c153c54/Snakefile#L42
3. User is using a profile that does not source from `parameters.yaml`, i.e. missing this line: https://github.com/nextstrain/ncov/blob/82ca8d48f671349c5688a055bcc67e108c153c54/nextstrain_profiles/nextstrain-open/config.yaml#L1-L2

    This seems the most likely for external usage, and I believe it's one of the reasons why we now recommend `--configfile` over `--profile` nowadays.

Reply:
> I think we need a broader pattern to adhere to here

The broader pattern seems to be having a default in `parameters.yaml` + direct access in Snakemake. This is how it's done for other config:

https://github.com/nextstrain/ncov/blob/83992903c54496513dcdf0b5acdef17d834c0ad5/workflow/snakemake_rules/main_workflow.smk#L1165-L1168

https://github.com/nextstrain/ncov/blob/83992903c54496513dcdf0b5acdef17d834c0ad5/defaults/parameters.yaml#L143-L152

Reply:
Thanks for the thoughts. I agree with your logic. Though I would like to push the fixes to a separate PR.